src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientOptionsAnalyzer.cs

```diff
@@ -45,6 +47,21 @@ private void AnalyzeClientOptionsType(ISymbolAnalysisContext context)
                 return;
             }
 
+            foreach (var serviceVersionMember in serviceVersionEnum.GetMembers().Where(member => member.Kind == SymbolKind.Field))
+            {
+                var parts = serviceVersionMember.Name.Split('_', StringSplitOptions.RemoveEmptyEntries);
```

Comment:
>StringSplitOptions.RemoveEmptyEntries
Maybe we shouldn't. `V__2020` would be a wrong enum member to have

Reply:
Worth allocating a separate descriptor? Or maybe just genericize the existing one I added? I hate to make things too generic, but in this case something like the following could work:

> Invalid ServiceVersion member name.
> All parts of ServiceVersion members' names must begin with a number or uppercase letter and cannot have consecutive underscores.

We could add to that sentence in the unlikely case we think of more things. Or do you prefer separate, more-specific descriptors?

Reply:
I think `All parts of ServiceVersion members' names must begin with a number or uppercase letter and cannot have consecutive underscores.` is great. I prefer more specific descriptors in general but this seems like a pretty simple error to describe.